### PR TITLE
Feature: Add GraphQL source plugin

### PR DESCRIPTION
**Pull request by jakedohm**
Here is my recommended implementation for the `graphql-source` plugin! It covers all of the bases, I believe: schema stitching, field namespacing, type conflict prevention, and allowing for custom auth with headers!

I'm super excited to see this land in core, and happy to review and updates/changes/nitpicks 😄

### CODE REVIEW

**Comment by jakedohm** [comment]
This relates to #80 😄

**Comment by hjvedvik** [comment]
Great work @jakedohm! I don't have much to comment. Just fix the linting and it should be okay :)

**Comment by hjvedvik** [comment]
Do you think `@gridsome/plugin-graphql` would be a better name for this plugin? Just thinking out loud here, since this plugin isn't actually sourcing any nodes and won't work exactly like the other source plugins (thinking about the new routes config etc.). But I don't have anything against `@gridsome/source-graphql` if it's more preferred :)

**Comment by jakedohm** [comment]
I don't have a strong opinion, but I think we should stick with `source-graphql`. It is "sourcing" your data, and that's what Gatsby uses, so it makes sense to follow that convention, IMO.

**Comment by jakedohm** [comment]
@hjvedvik linting errors fixed, should be ready to merge!

File: packages/source-graphql/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "version": "1.0.0",
```
Comment (hjvedvik) [inline packages/source-graphql/package.json]:
```suggestion
  "version": "0.0.0",
```
We'll probably switch to `1.0.0` when Gridsome reaches v1.0 :)

**Comment by jakedohm** [comment]
@hjvedvik all requested changes have been made!

**Pull request merged**